pkg/cliplugins/workload/plugin/sync.go

```diff
@@ -248,14 +248,17 @@ func (o *SyncOptions) Run(ctx context.Context) error {
 	// cluster configuration since they only operate against a single workspace.
 	serverURL := configURL.Scheme + "://" + configURL.Host
 	input := templateInput{
-		ServerURL:                           serverURL,
-		CAData:                              base64.StdEncoding.EncodeToString(config.CAData),
-		Token:                               token,
-		KCPNamespace:                        o.KCPNamespace,
-		Namespace:                           o.DownstreamNamespace,
-		LogicalCluster:                      currentClusterName.String(),
-		SyncTarget:                          o.SyncTargetName,
-		SyncTargetUID:                       syncTargetUID,
+		ServerURL:    serverURL,
+		CAData:       base64.StdEncoding.EncodeToString(config.CAData),
+		Token:        token,
+		KCPNamespace: o.KCPNamespace,
+		Namespace:    o.DownstreamNamespace,
+
+		// run the syncer against a logical cluster, not the path, to make it resilient to workspace moves
+		SyncTargetPath: logicalcluster.From(syncTarget).Path().String(),
```

Comment:
To make this less confusing when reading the comment, omit the `Path()` call? 😄

Reply:
removed the comment :D